app/controllers/items_controller.rb

```diff
@@ -5,5 +5,6 @@ def index
 
   def show
     @item = Item.visible.find(params[:id])
+    @cart_item = @item.cart_items.build
```

Comment:
```suggestion
    @cart_item = current_cart.cart_items.find_or_initialize(item_id: params[:id])
```
こうすると、
- カートにitemが入ってなければbuildされて
- 入っていればfindされたcart_itemか返ってくる
ので、　viewでcreate/updateを分岐できるのでは？

Reply:
viewで場合分けするときの条件分岐の例
```haml
if cart_item.persisted?
  updateのとき
else
  createのとき
end
```

Reply:
viewでわけずに、controllerのcreateで `find_or_create`を使って、1.新規作成と2.更新（quantityをupdate）を分岐させるほうが今回は自然かもしれない。